components/workflow-mgt/org.wso2.carbon.identity.workflow.impl/src/main/java/org/wso2/carbon/identity/workflow/impl/WorkflowImplServiceImpl.java

```diff
@@ -122,6 +159,103 @@ public void deleteHumanTask(WorkflowRequest workflowRequest) throws WorkflowImpl
         }
     }
 
+    /**
+     * This method is used to remove the BPS Artifacts upon a deletion of
+     * a Workflow.
+     *
+     * @param workflow - Workflow request to be deleted.
+     * @throws WorkflowImplException
+     */
+
+    @Override
+    public void removeBPSPackage(Workflow workflow) throws WorkflowImplException {
+
+        WorkflowImplService workflowImplService = WorkflowImplServiceDataHolder.getInstance().getWorkflowImplService();
+        WorkflowManagementService workflowManagementService = WorkflowImplServiceDataHolder.getInstance().
+                getWorkflowManagementService();
+
+        if (workflowImplService == null || workflowManagementService == null) {
+            throw new WorkflowImplException("Error while deleting the BPS artifacts of: " + workflow.getWorkflowName());
+        }
+
+        try {
+            List<Parameter> workflowParameters = workflowManagementService.
+                    getWorkflowParameters(workflow.getWorkflowId());
+            Parameter bpsParameter = WorkflowManagementUtil.getParameter(workflowParameters,
+                    WFImplConstant.ParameterName.BPS_PROFILE, WFConstant.ParameterHolder.WORKFLOW_IMPL);
+            if (bpsParameter == null) {
+                throw new WorkflowImplException("Error while deleting the BPS artifacts of: " +
+                        workflow.getWorkflowName());
+            }
+            String bpsProfileName = bpsParameter.getParamValue();
+            if (StringUtils.isEmpty(bpsProfileName)) {
+                throw new WorkflowImplException("Error while deleting the BPS artifacts of: " +
+                        workflow.getWorkflowName());
+            }
+
+            int tenantId = CarbonContext.getThreadLocalCarbonContext().getTenantId();
+            BPSProfile bpsProfile = workflowImplService.getBPSProfile(bpsProfileName, tenantId);
+
+            if (log.isDebugEnabled()) {
+                log.debug("Removing BPS Artifacts of" + " " + bpsProfileName + " " + "for Tenant ID : " + tenantId);
```

Comment:
Merge adjacent string literals in same line
eg. "Removing BPS Artifacts of" + " " can be joined to "Removing BPS Artifacts of "